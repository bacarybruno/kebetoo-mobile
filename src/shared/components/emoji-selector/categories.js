const categories = {
  history: {
    symbol: '🕘',
    name: 'Recently used',
  },
  emotion: {
    symbol: '😀',
    name: 'Smileys & Emotion',
  },
  nature: {
    symbol: '🐻',
    name: 'Animals & Nature',
  },
  food: {
    symbol: '🍔',
    name: 'Food & Drink',
  },
  activities: {
    symbol: '⚽',
    name: 'Activities',
  },
  places: {
    symbol: '✈️',
    name: 'Travel & Places',
  },
  objects: {
    symbol: '💡',
    name: 'Objects',
  },
  symbols: {
    symbol: '🔣',
    name: 'Symbols',
  },
  flags: {
    symbol: '🏳️‍',
    name: 'Flags',
  },
}

const categoryNames = Object.values(categories).map((category) => category.name)
const categoryIcons = Object.values(categories).map((category) => category.symbol)
const fullCategoryNames = [...categoryNames]
fullCategoryNames.splice(2, 0, 'People & Body')

export {
  categoryNames,
  categoryIcons,
  fullCategoryNames,
}
export default categories
