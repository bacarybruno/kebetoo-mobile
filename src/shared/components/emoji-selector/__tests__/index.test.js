import { act } from 'react-test-renderer'

import setupTest from '@app/config/jest-setup'

import EmojiSelector, { Emoji, EmojiTab } from '../index'

import { categoryNames } from '../categories'

const storeState = {
  appReducer: {
    emojiHistory: [{
      count: 1, symbol: '😀', addedOn: Date.now(),
    }, {
      count: 2, symbol: '🦁', addedOn: Date.now(),
    }],
  },
}

const emptyStoreState = {
  appReducer: {
    emojiHistory: [],
  },
}

const givenEmojiSelector = setupTest(EmojiSelector)({
  onSelectItem: jest.fn(),
  __storeState__: storeState,
})

const givenEmptyStateEmojiSelector = setupTest(EmojiSelector)({
  onSelectItem: jest.fn(),
  __storeState__: emptyStoreState,
})

it('renders EmojiSelector', () => {
  const { wrapper } = givenEmojiSelector()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('renders EmojiSelector without history', () => {
  const { wrapper } = givenEmptyStateEmojiSelector()
  expect(wrapper.root.findByProps({ testID: 'emoji-list' }).props.initialScrollIndex).toBe(1)
})

it('renders all tabs headers', () => {
  const { wrapper } = givenEmojiSelector()
  expect(wrapper.root.findAllByType(EmojiTab).length).toBe(categoryNames.length)
})

it('switches tabs', () => {
  const { wrapper } = givenEmojiSelector()
  const tabIndex = 1
  act(() => {
    wrapper.root.findByProps({ testID: `emoji-tab-${tabIndex}` }).props.onPress()
  })
  expect(wrapper.root.findByProps({ testID: 'emoji-list' }).props.initialScrollIndex).toBe(tabIndex)
})

it('selects emoji', () => {
  const { wrapper, props } = givenEmojiSelector()
  wrapper.root.findAllByType(Emoji)[0].props.onPress('😀')
  expect(props.onSelectItem).toBeCalledTimes(1)
  expect(props.onSelectItem).toBeCalledWith('😀')
})
