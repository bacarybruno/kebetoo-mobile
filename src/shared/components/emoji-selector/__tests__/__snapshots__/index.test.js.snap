// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EmojiSelector 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "zIndex": 1,
        },
        false,
      ]
    }
  >
    <View
      accessibilityElementsHidden={false}
      importantForAccessibility="auto"
      onLayout={[Function]}
      pointerEvents="box-none"
      style={null}
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "borderBottomColor": "rgb(216, 216, 216)",
              "flex": 1,
              "shadowColor": "rgb(216, 216, 216)",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0,
              },
              "shadowOpacity": 0.85,
              "shadowRadius": 0,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "maxHeight": undefined,
            "minHeight": undefined,
            "opacity": undefined,
            "transform": undefined,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 0,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "marginHorizontal": 16,
                "opacity": 1,
              }
            }
          >
            <Text
              accessibilityRole="header"
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 17,
                  "fontWeight": "600",
                }
              }
            >
              jest
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
      <View
        accessibilityElementsHidden={false}
        closing={false}
        gestureVelocityImpact={0.3}
        importantForAccessibility="auto"
        onClose={[Function]}
        onGestureBegin={[Function]}
        onGestureCanceled={[Function]}
        onGestureEnd={[Function]}
        onOpen={[Function]}
        onTransitionStart={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        transitionSpec={
          Object {
            "close": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
            "open": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            forwardedRef={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            style={
              Object {
                "flex": 1,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": -1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                  "top": 0,
                  "width": 3,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#F5F5F5",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "#F5F5F5",
                              "borderBottomWidth": 2,
                              "flex": 1,
                              "height": 40,
                              "justifyContent": "center",
                            },
                            Object {
                              "borderBottomColor": "rgba(0, 122, 255, 255)",
                            },
                          ]
                        }
                        testID="emoji-tab-0"
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 20,
                            }
                          }
                        >
                          🕘
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "#F5F5F5",
                              "borderBottomWidth": 2,
                              "flex": 1,
                              "height": 40,
                              "justifyContent": "center",
                            },
                            false,
                          ]
                        }
                        testID="emoji-tab-1"
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 20,
                            }
                          }
                        >
                          😀
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "#F5F5F5",
                              "borderBottomWidth": 2,
                              "flex": 1,
                              "height": 40,
                              "justifyContent": "center",
                            },
                            false,
                          ]
                        }
                        testID="emoji-tab-2"
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 20,
                            }
                          }
                        >
                          🐻
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "#F5F5F5",
                              "borderBottomWidth": 2,
                              "flex": 1,
                              "height": 40,
                              "justifyContent": "center",
                            },
                            false,
                          ]
                        }
                        testID="emoji-tab-3"
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 20,
                            }
                          }
                        >
                          🍔
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "#F5F5F5",
                              "borderBottomWidth": 2,
                              "flex": 1,
                              "height": 40,
                              "justifyContent": "center",
                            },
                            false,
                          ]
                        }
                        testID="emoji-tab-4"
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 20,
                            }
                          }
                        >
                          ⚽
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "#F5F5F5",
                              "borderBottomWidth": 2,
                              "flex": 1,
                              "height": 40,
                              "justifyContent": "center",
                            },
                            false,
                          ]
                        }
                        testID="emoji-tab-5"
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 20,
                            }
                          }
                        >
                          ✈️
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "#F5F5F5",
                              "borderBottomWidth": 2,
                              "flex": 1,
                              "height": 40,
                              "justifyContent": "center",
                            },
                            false,
                          ]
                        }
                        testID="emoji-tab-6"
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 20,
                            }
                          }
                        >
                          💡
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "#F5F5F5",
                              "borderBottomWidth": 2,
                              "flex": 1,
                              "height": 40,
                              "justifyContent": "center",
                            },
                            false,
                          ]
                        }
                        testID="emoji-tab-7"
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 20,
                            }
                          }
                        >
                          🔣
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "borderBottomColor": "#F5F5F5",
                              "borderBottomWidth": 2,
                              "flex": 1,
                              "height": 40,
                              "justifyContent": "center",
                            },
                            false,
                          ]
                        }
                        testID="emoji-tab-8"
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 20,
                            }
                          }
                        >
                          🏳️‍
                        </Text>
                      </View>
                    </View>
                    <RCTScrollView
                      data={
                        Array [
                          "Recently used",
                          "Smileys & Emotion",
                          "Animals & Nature",
                          "Food & Drink",
                          "Activities",
                          "Travel & Places",
                          "Objects",
                          "Symbols",
                          "Flags",
                        ]
                      }
                      disableVirtualization={false}
                      getItem={[Function]}
                      getItemCount={[Function]}
                      getItemLayout={[Function]}
                      horizontal={true}
                      initialNumToRender={1}
                      initialScrollIndex={0}
                      keyExtractor={[Function]}
                      keyboardShouldPersistTaps="handled"
                      maxToRenderPerBatch={10}
                      onContentSizeChange={[Function]}
                      onEndReachedThreshold={2}
                      onLayout={[Function]}
                      onMomentumScrollEnd={[Function]}
                      onScroll={[Function]}
                      onScrollBeginDrag={[Function]}
                      onScrollEndDrag={[Function]}
                      pagingEnabled={true}
                      removeClippedSubviews={true}
                      renderItem={[Function]}
                      scrollEventThrottle={50}
                      stickyHeaderIndices={Array []}
                      testID="emoji-list"
                      updateCellsBatchingPeriod={50}
                      viewabilityConfigCallbackPairs={Array []}
                      windowSize={3}
                    >
                      <View>
                        <View
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <RCTScrollView
                            ListEmptyComponent={
                              Object {
                                "$$typeof": Symbol(react.memo),
                                "compare": null,
                                "type": [Function],
                              }
                            }
                            data={
                              Array [
                                Object {
                                  "addedOn": 1593084600000,
                                  "count": 1,
                                  "symbol": "😀",
                                },
                                Object {
                                  "addedOn": 1593084600000,
                                  "count": 2,
                                  "symbol": "🦁",
                                },
                              ]
                            }
                            disableVirtualization={false}
                            getItem={[Function]}
                            getItemCount={[Function]}
                            getItemLayout={[Function]}
                            horizontal={false}
                            initialNumToRender={10}
                            keyExtractor={[Function]}
                            keyboardShouldPersistTaps="handled"
                            maxToRenderPerBatch={10}
                            onContentSizeChange={[Function]}
                            onEndReachedThreshold={2}
                            onLayout={[Function]}
                            onMomentumScrollEnd={[Function]}
                            onScroll={[Function]}
                            onScrollBeginDrag={[Function]}
                            onScrollEndDrag={[Function]}
                            removeClippedSubviews={true}
                            renderItem={[Function]}
                            scrollEventThrottle={50}
                            stickyHeaderIndices={Array []}
                            updateCellsBatchingPeriod={50}
                            viewabilityConfigCallbackPairs={Array []}
                            windowSize={8}
                          >
                            <View>
                              <View
                                style={null}
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "flexDirection": "row",
                                      },
                                      Object {
                                        "width": 750,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    accessible={true}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "height": 93.75,
                                        "justifyContent": "center",
                                        "width": 93.75,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Object {
                                          "fontSize": 27,
                                        }
                                      }
                                    >
                                      😀
                                    </Text>
                                  </View>
                                  <View
                                    accessible={true}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "height": 93.75,
                                        "justifyContent": "center",
                                        "width": 93.75,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Object {
                                          "fontSize": 27,
                                        }
                                      }
                                    >
                                      🦁
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                        <View
                          style={
                            Object {
                              "width": 6000,
                            }
                          }
                        />
                      </View>
                    </RCTScrollView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
